### PR TITLE
Fix unexpected node crash when clock is behind

### PR DESCRIPTION
**Pull request by pavel-kokolemin**

### CODE REVIEW

File: chainstate/test-suite/src/tests/syncing_tests.rs [hunk 1]
```diff
@@ -490,3 +490,52 @@ fn initial_block_download(#[case] seed: Seed) {
         assert!(!tf.chainstate.is_initial_block_download().unwrap());
     });
 }
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn orphan_block(#[case] seed: Seed) {
```
Comment (TheQuantumPhysicist) [inline chainstate/test-suite/src/tests/syncing_tests.rs:497]: header_check_for_orphan

**Review by TheQuantumPhysicist — APPROVED** [review]

**Review by iljakuklic — APPROVED** [review]

File: chainstate/test-suite/src/tests/syncing_tests.rs [hunk 2]
```diff
@@ -490,3 +490,52 @@ fn initial_block_download(#[case] seed: Seed) {
...
 }
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn header_check_for_orphan(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut rng = make_seedable_rng(seed);
+        let mut tf = TestFramework::builder(&mut rng)
+            .with_chainstate_config(ChainstateConfig {
```
Comment (discosultan) [inline chainstate/test-suite/src/tests/syncing_tests.rs:501]: Should we shorten this to `ChainstateConfig::default()`?

Comment (pavel-kokolemin) [reply chainstate/test-suite/src/tests/syncing_tests.rs:501]: @TheQuantumPhysicist asked to explicitly list all the fields to see all the places where adding a new field might have an effect.

**Review by discosultan — APPROVED** [review]
